### PR TITLE
Feature/local messaging service

### PR DESCRIPTION
**Pull request by rodrigopavezi**
@tejohnso @stulees please review. I tried to do an unit test mocking ipc with simple-mock but I could not. It does not mock the ipc object. I will leave it for now until we find a way to do that. cheers

### CODE REVIEW

**Comment by rodrigopavezi** [comment]
@tejohnso I think they also need to receive messages that is why I created a client to expose both methods. In this way it is pretty hand to include it on modules. You can see also that the client is singleton so it want make multiple connections to LMS. cheers

**Comment by stulees** [comment]
@rodrigopavezi By making it a singleton, won't this restrict from applying a different value for `ipc.config.id` when each client connects?

**Comment by rodrigopavezi** [comment]
@stulees I guess we should not have a single module being 2 clients at the same time. Does that make sense? cheers

**Comment by stulees** [comment]
@rodrigopavezi Or the `ipc.config.id` should be generic and not be passed in as a param. A client will apply a value for `from` in the message that they need to broadcast anyway.

**Comment by rodrigopavezi** [comment]
@stulees I guess we could I just don't think there is harm on leaving it there. We might use it for logging. cheers

**Comment by stulees** [comment]
@rodrigopavezi So when all modules call `getLMSClient(id)`, what is the value of `id` they should pass in?

**Comment by rodrigopavezi** [comment]
@stulees they should pass the ID of the module, such as player or installer. That will create a singleton client for the entire module so the module does not make multiple connections to the local message service. cheers

**Comment by rodrigopavezi** [comment]
@stulees here is how it is used on player
```
function broadcastMessage(message) {
  moduleCommon.getLMSClient("player").then((lmsClient)=>{
    lmsClient.broadcastMessage(message);
  });
}
```

cheers

**Comment by stulees** [comment]
@rodrigopavezi Ah, I follow you now. So given that, I agree on exposing receive method as well as broadcast, that is handy. I also agree with @tejohnso on not exposing the LMS module name in the API. Can we change `getLMSClient` to `connect`?

**Comment by rodrigopavezi** [comment]
@stulees sure, changed. cheers

File: test/integration/lms.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+const assert = require("assert");
+const ipc = require('node-ipc');
+const config = require("../../common.js");
+
+describe("LMS", ()=>{
+  describe("getLMSClient", ()=>{
```
Comment (stulees) [inline test/integration/lms.js]: @rodrigopavezi Can you change to "connect" and is the test possible to be put in `integration/common.js`?

Comment (rodrigopavezi) [reply test/integration/lms.js]: @stulees sure, done. cheers

**Comment by rodrigopavezi** [comment]
@stulees please review latest commit. cheers

**Comment by stulees** [comment]
@rodrigopavezi 👍

**Comment by tejohnso** [comment]
@rodrigopavezi I don't think you need a promise returned in your *receiveMessages()* since the **EE** can be returned immediately.

**Comment by rodrigopavezi** [comment]
@tejohnso makes sense. I will change. cheers

**Pull request merged**